packages/backend-storage/src/factory.ts

```diff
@@ -52,7 +58,22 @@ export class AmplifyStorageFactory
     const amplifyStorage = constructContainer.getOrCompute(
       this.generator
     ) as AmplifyStorage;
-    Aspects.of(Stack.of(amplifyStorage)).add(new StorageValidator());
+
+    const firstStorage = Stack.of(amplifyStorage).node.children.filter(
+      (el) => el instanceof AmplifyStorage
+    )[0] as AmplifyStorage;
+
+    /*
+     * only call Aspects once,
+     * otherwise there will be the an error -
+     * "there is already a construct with name 'storageRegion'"
+     */
+    if (firstStorage.name === this.props.name) {
```

Comment:
Had to handle it in this hacky way to make sure the Aspect only runs once. 

Would it be better to handle it on the top stack level? But that'll be out of the storage package?

Any suggestions?

Reply:
you can try something like 

```
const aspects = Aspects.of(Stack.of(amplifyStorage));
if (!aspects.all.find(aspect => aspect instanceof StorageValidator) { // or any other predicate that would work here.
  aspects.add(new StorageValidator(...));
}
```